index.js

```diff
@@ -48,7 +48,7 @@ async function fastifyEtag (app, opts) {
       reply.header('etag', etag)
     }
 
-    if (req.headers['if-none-match'] === etag) {
+    if (req.headers['if-none-match'] === etag || req.headers['if-none-match'] === 'W/' + etag || 'W/' + req.headers['if-none-match'] === etag) {
```

Comment:
I have somehow the feeling that we can optimize this.

we have opts.weak so we know if our generated etag is weak or not.

Reply:
Please see the issue: https://github.com/fastify/fastify-etag/issues/108

Reply:
but you have etag. So you could check if it is a WeakTag. like 

```javascript
function isWeakTag(value) {
  return (
    value[0] === 'W' &&
    value[1] === '\'
  )
}
```

```suggestion
    const ifNoneMatch = req.headers['if-none-match']

    if (ifNoneMatch === etag || (isWeakTag(etag) ? 'W/' + ifNoneMatch === etag : ifNoneMatch === 'W/' + etag)) {
```

Reply:
I will benchmark and see if it has performance improvements